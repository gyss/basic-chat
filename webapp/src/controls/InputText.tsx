/** @jsx jsx */

import * as React from 'react'
import {css, jsx} from '@emotion/core'

const inputStyles = css`
  max-width: 100%;
  width: 100%;
  background-color: #fff;
  border-color: #dbdbdb;
  border: 1px solid transparent;
  border-radius: 4px;
  display: inline-flex;
  font-size: 1rem;
  height: 2em;
  line-height: 1.5;
  padding-left: calc(.75em - 1px);
  padding-right: calc(.75em - 1px);
`

export default function InputText(props: React.InputHTMLAttributes<HTMLInputElement>) {
  return (
    <input css={inputStyles} {...props} />
  )
}