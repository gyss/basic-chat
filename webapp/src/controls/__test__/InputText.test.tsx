/** @format */

import * as React from 'react'
import {render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import InputText from '../InputText'

describe('<InputText>', () => {
  test('should render', () => {
    const {asFragment} = render(<InputText name="username" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
