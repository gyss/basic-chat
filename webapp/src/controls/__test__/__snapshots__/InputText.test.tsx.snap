// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputText> should render 1`] = `
<DocumentFragment>
  <input
    class="css-jkzcq5-InputText"
    name="username"
    type="text"
  />
</DocumentFragment>
`;
